client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,339 @@
+        settings = project_settings["deadline"]
+        profiles = settings["publish"][cls.__name__]["profiles"]
+
+        cls.profiles = profiles or []
+
+        addons_manager = AddonsManager(project_settings)
+        deadline_addon = addons_manager["deadline"]
+        deadline_server_name = settings["deadline_server"]
+        pools = deadline_addon.get_pools_by_server_name(deadline_server_name)
+        for pool in pools:
+            cls.pool_enum_values.append({"value": pool, "label": pool})
+
+        groups = deadline_addon.get_groups_by_server_name(deadline_server_name)
+        for group in groups:
+            cls.group_enum_values.append({"value": group, "label": group})
+
+        limit_groups = (
+            deadline_addon.get_limit_groups_by_server_name(deadline_server_name))
+        if not limit_groups:
+            limit_groups.append("none")  # enum cannot be empty
+        for limit_group in limit_groups:
+            cls.limit_group_enum_values.append(
+                {"value": limit_group, "label": limit_group})
+
+        machines = (
+            deadline_addon.get_machines_by_server_name(deadline_server_name))
+        for machine in machines:
+            cls.machines_enum_values.append(
+                {"value": machine, "label": machine})
+
+    @classmethod
+    def get_attr_defs_for_instance(cls, create_context, instance):
+        """Get list of attr defs that are set in Settings as artist overridable
+
+        Args:
+            create_context (ayon_core.pipeline.create.CreateContext)
+            instance (ayon_core.pipeline.create.CreatedInstance):
+
+        Returns:
+            (list)
+        """
+        if instance.product_type not in cls.families:
+            return []
+
+        host_name = create_context.host_name
+
+        task_name = instance["task"]
+        folder_path = instance["folderPath"]
+        task_entity = create_context.get_task_entity(folder_path, task_name)
+
+        task_name = task_type = None
+        if task_entity:
+            task_name = task_entity["name"]
+            task_type = task_entity["taskType"]
+        profile = filter_profiles(
+            cls.profiles,
+            {
+                "host_names": host_name,
+                "task_types": task_type,
+                "task_names": task_name,
+                # "product_type": product_type
+            }
+        )
+        if not profile:
+            return []
+        overrides = set(profile["overrides"])
+        if not overrides:
+            return []
+
+        defs = [
+            UISeparatorDef("deadline_defs_starts"),
+        ]
+
+        defs.extend(cls._get_artist_overrides(overrides, profile))
+
+        # explicit frames to render - for test renders
+        defs.append(
+            TextDef(
+                "frames",
+                label="Frames",
+                default="",
+                tooltip="Explicit frames to be rendered. (1, 3-4)"
+            )
+        )
```

Comment:
Yup - there will be cases an instance may not have the attributes shown if it didn't initially have the publish families explicitly set.

Reply:
I checked it again.
so https://github.com/ynput/ayon-houdini/pull/164 doesn't solve the issue. 

The issue: currently my pointcache instance in Houdini doesn't have the attributes in the publisher UI and this particular attribute `frames` is missing because it is not in the profile or in the attributes of the instance. 
which leads to the missing key error: 
here's some log (I printed the `attr_values` before the error occurs.)
```python
DEBUG: attr_values: {'host_names': [], 'task_types': [], 'task_names': [], 'chunk_size': 999, 'priority': 50, 'group': '', 'limit_groups': [], 'primary_pool': '', 'secondary_pool': '', 'machine_limit': 0, 'machine_list': [], 'machine_list_deny': False, 'concurrent_tasks': 1, 'department': '', 'job_delay': '', 'use_published': True, 'use_asset_dependencies': True, 'use_workfile_dependency': True, 'multiprocess': False, 'additional_job_info': '', 'additional_plugin_info': '', 'overrides': ['department', 'chunk_size', 'group', 'priority', 'primary_pool', 'secondary_pool']}
Traceback (most recent call last):
  File "C:\Users\Mustafa Taher\AppData\Local\Ynput\AYON\dependency_packages\ayon_2411052009_windows.zip\dependencies\pyblish\plugin.py", line 528, in __explicit_process
    runner(*args)
  File "E:\Ynput\ayon-deadline\client\ayon_deadline\plugins\publish\global\collect_jobinfo.py", line 53, in process
    job_info = AYONDeadlineJobInfo.from_dict(attr_values)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Ynput\ayon-deadline\client\ayon_deadline\lib.py", line 524, in from_dict
    "Frames": data["frames"],
              ~~~~^^^^^^^^^^
KeyError: 'frames'
```

Reply:
fixed https://github.com/ynput/ayon-deadline/pull/49/commits/c05ddc2d39f340c6d84e7c2876d3cd6118d8be3f